core/src/main/java/tech/pegasys/eth2signer/core/multikey/SigningMetadataConfigLoader.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.eth2signer.core.multikey;
+
+import tech.pegasys.eth2signer.core.multikey.metadata.SigningMetadataFile;
+import tech.pegasys.eth2signer.core.multikey.metadata.SigningMetadataFileProvider;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class SigningMetadataConfigLoader {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Path configsDirectory;
+  private SigningMetadataFileProvider metadataFileProvider;
+  private String fileExtension;
+
+  public SigningMetadataConfigLoader(
+      final Path rootDirectory,
+      final String fileExtension,
+      final SigningMetadataFileProvider metadataFileProvider) {
+    this.configsDirectory = rootDirectory;
+    this.metadataFileProvider = metadataFileProvider;
+    this.fileExtension = fileExtension;
+  }
+
+  Optional<SigningMetadataFile> loadMetadataForAddress(final String address) {
```

Comment:
few things:
1. does this need explicit visibility (public)?
2. Can we abstract away from "address"? I.e. eth2signer doens't know about addresses ... its more identifiers (I think?)
3. There's a smell, that this returns a SigningMetadataFILE - but is a loader ... shouldn't it provide just SigningMetadata? Either the return type is not quite right, or the name of the function is more "getMetadataFile(final String identifier)

Reply:
1. It's got package level access, don't see how we can restrict this any further
2. Think it's more public keys than addresses? Would that be better?
3. I guess it is a bit odd. This is existing code, didn't think too much about the code I copied.